core/query/filter/typecheck_test.go

```diff
@@ -1,15 +1,31 @@
 package filter
 
-import "testing"
+import (
+	"errors"
+	"testing"
+
+	"chain/testutil"
+)
 
 func TestTypeCheckInvalid(t *testing.T) {
 	testCases := []struct {
-		p string
+		p   string
+		err error
 	}{
-		{p: `1 = 'hello world'`},
-		{p: `INPUTS('hello')`},
-		{p: `foo(1=1).bar`},
-		{p: `'hello'.foo`},
+		{p: `1 = 'hello world'`, err: errors.New("= expects operands of matching types")},
+		{p: `inputs('hello')`, err: errors.New("inputs(...) body must have type bool")},
+		{p: `inputs(1=1).bar`, err: errors.New("selector `.` can only be used on objects")},
+		{p: `'hello'.foo`, err: errors.New("selector `.` can only be used on objects")},
+		{p: `inputs(amount = asset_id)`, err: errors.New("= expects operands of matching types")},
+		{p: `inputs(100 = asset_id)`, err: errors.New("= expects operands of matching types")},
+		{p: `inputs(wat)`, err: errors.New("invalid attribute: wat")},
+		{p: `wat(asset_id = 'a')`, err: errors.New("invalid environment `wat`")},
+		{p: `position(asset_id = 'a')`, err: errors.New("invalid environment `position`")},
+		{p: `('a' = 'a') = (1 = 1)`, err: errors.New("= expects integer or string operands")},
```

Comment:
Yeah, I think we will want to relax this soon, but it's tricky at the moment because the booleans that we expose on our own objects are exposed as strings. I think we should figure out what to do with those fields before relaxing it.